bitcoin/src/blockdata/transaction.rs

```diff
@@ -714,11 +714,16 @@ impl Transaction {
     #[deprecated(since = "0.31.0", note = "Use Transaction::stripped_size() instead")]
     pub fn strippedsize(&self) -> usize { Self::stripped_size(self).to_wu() as usize }
 
+    // TODO this should return a type that expresses size in terms of witness vs non-witness.
+    // struct Weight {
+    //     non_witness_bytes: u64,
+    //     witness_bytes: u64,
+    // }
     /// Returns the size of this transaction excluding the witness data.
     pub fn stripped_size(&self) -> Weight {
         let mut input_size: Weight = Weight::ZERO;
         for input in &self.input {
-            input_size += TxIn::BASE_WEIGHT
+            input_size += Weight::from_wu(32 + 4 + 4)
```

Comment:
I'm seriously sleep deprived right now but I don't see why these magic numbers are weight units and above they are virtual bytes. Can you add code comments to justify the magic numbers please.